### ISSUE TITLE
Some confusing points

### ISSUE THREAD

**Issue by wjk1008** [zjunlp/EasyEdit#110]
1. It seems to lack "load_from_pretrained(q_former_model)" in the init of Blip2OPT(https://github.com/zjunlp/EasyEdit/blob/main/easyeditor/trainer/blip2_models/blip2_opt.py)
2. In https://github.com/zjunlp/EasyEdit/blob/main/hparams/TRAINING/SERAC/blip2.yaml, why qformer_checkpoint is blip2_pretrained_flant5xxl.pth instead of blip2_pretrained_opt2.7B.pth as the model is BLIPOPT.
3. In the data pre-processing of MM_EDIT(https://github.com/zjunlp/EasyEdit/blob/main/easyeditor/dataset/coco_caption.py or https://github.com/zjunlp/EasyEdit/blob/main/easyeditor/dataset/vqa.py), you zip prompt and target together and feed them into the model, but in testing, will this cause label leaking?
Thanks for your attention.

**Comment by cheng-siyuan** [comment]
Thank you for being so interested in our paper!
1, 2. This paper is an empirical analysis of editing a multimodal large model, here is a bug, thank you very much for finding it, after preliminary testing, it does not affect the experimental conclusions. The results will be updated after re-running and updating the GitHub checkpoint.
3. There will be no label leaking involved here, it is a feature of model editing. The knowledge you want to update in the model, you need to make the model aware of where to direct the update (a.k.a. golden label).

**Comment by wjk1008** [comment]
> Thank you for being so interested in our paper! 1, 2. This paper is an empirical analysis of editing a multimodal large model, here is a bug, thank you very much for finding it, after preliminary testing, it does not affect the experimental conclusions. The results will be updated after re-running and updating the GitHub checkpoint. 3. There will be no label leaking involved here, it is a feature of model editing. The knowledge you want to update in the model, you need to make the model aware of where to direct the update (a.k.a. golden label).

Sorry, I still don't understand, in the process of editing,  the target is needed. But in the evaluating, we have to compare the logits between the origin model and edited model. Is it reasonable to feed the model prompt and target together to get the logits?

**Comment by wjk1008** [comment]
Thank you for your reply. Interested in your work, I have two more questions.
1. How can I reproduce your results in the paper "Can We Edit Multimodal Large Language Models".  More precisely, is the "result.json" generated by test functions in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py the same metrics in the paper?
2.  When I run train_MEND_Blip2OPT_Caption in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py, loss always is nan, how can I solve it？

**Comment by cheng-siyuan** [comment]
> > Thank you for being so interested in our paper! 1, 2. This paper is an empirical analysis of editing a multimodal large model, here is a bug, thank you very much for finding it, after preliminary testing, it does not affect the experimental conclusions. The results will be updated after re-running and updating the GitHub checkpoint. 3. There will be no label leaking involved here, it is a feature of model editing. The knowledge you want to update in the model, you need to make the model aware of where to direct the update (a.k.a. golden label).
> 
> Sorry, I still don't understand, in the process of editing, the target is needed. But in the evaluating, we have to compare the logits between the origin model and edited model. Is it reasonable to feed the model prompt and target together to get the logits?

I interpret the logits you are comparing here as a metric of locality. We only need to keep the logits stable before and after editing in order to maintain the rest of the knowledge of the model in our experiments. On the other hand, it's a unidirectional attention model putting together prompt and target to predict that does not have a leakage problem.

**Comment by tbozhong** [comment]
> Thank you for your reply. Interested in your work, I have two more questions.
> 
> 1. How can I reproduce your results in the paper "Can We Edit Multimodal Large Language Models".  More precisely, is the "result.json" generated by test functions in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py the same metrics in the paper?
> 2. When I run train_MEND_Blip2OPT_Caption in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py, loss always is nan, how can I solve it？

1. We will update our BLIP-2 code later, and you can first reproduce the MiniGPT-4 editing experiment. Second, we make sure that the metrics are consistent with those in the paper.
2. We were running it before without nan issues, we are rechecking it, please wait for the following update.

**Comment by wjk1008** [comment]
> > Thank you for your reply. Interested in your work, I have two more questions.
> > 
> > 1. How can I reproduce your results in the paper "Can We Edit Multimodal Large Language Models".  More precisely, is the "result.json" generated by test functions in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py the same metrics in the paper?
> > 2. When I run train_MEND_Blip2OPT_Caption in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py, loss always is nan, how can I solve it？
> 
> 1. We will update our BLIP-2 code later, and you can first reproduce the MiniGPT-4 editing experiment. Second, we make sure that the metrics are consistent with those in the paper.
> 2. We were running it before without nan issues, we are rechecking it, please wait for the following update.

Thanks for your reply. For the question 1, my expression may be unclear. My question is the "result.json" generated by test functions in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py, either BLIP2 or MiniGPT4, is whether or not the final results like the metrics in the table2 of "Can We Edit Multimodal Large Language Models". In other words, if I want to reproduce the results, Do I only need to run a train_{algorithm}_{model}_{task} function and run a test_{algorithm}_{model}_{task} function, and then see the result.json?

**Comment by wjk1008** [comment]
> > > Thank you for being so interested in our paper! 1, 2. This paper is an empirical analysis of editing a multimodal large model, here is a bug, thank you very much for finding it, after preliminary testing, it does not affect the experimental conclusions. The results will be updated after re-running and updating the GitHub checkpoint. 3. There will be no label leaking involved here, it is a feature of model editing. The knowledge you want to update in the model, you need to make the model aware of where to direct the update (a.k.a. golden label).
> > 
> > 
> > Sorry, I still don't understand, in the process of editing, the target is needed. But in the evaluating, we have to compare the logits between the origin model and edited model. Is it reasonable to feed the model prompt and target together to get the logits?
> 
> I interpret the logits you are comparing here as a metric of locality. We only need to keep the logits stable before and after editing in order to maintain the rest of the knowledge of the model in our experiments. On the other hand, it's a unidirectional attention model putting together prompt and target to predict that does not have a leakage problem.

Thanks for explaining it to me, now I get it.

**Comment by tbozhong** [comment]
> > > Thank you for your reply. Interested in your work, I have two more questions.
> > > 
> > > 1. How can I reproduce your results in the paper "Can We Edit Multimodal Large Language Models".  More precisely, is the "result.json" generated by test functions in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py the same metrics in the paper?
> > > 2. When I run train_MEND_Blip2OPT_Caption in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py, loss always is nan, how can I solve it？
> > 
> > 
> > 
> > 1. We will update our BLIP-2 code later, and you can first reproduce the MiniGPT-4 editing experiment. Second, we make sure that the metrics are consistent with those in the paper.
> > 2. We were running it before without nan issues, we are rechecking it, please wait for the following update.
> 
> Thanks for your reply. For the question 1, my expression may be unclear. My question is the "result.json" generated by test functions in https://github.com/zjunlp/EasyEdit/blob/main/multimodal_edit.py, either BLIP2 or MiniGPT4, is whether or not the final results like the metrics in the table2 of "Can We Edit Multimodal Large Language Models". In other words, if I want to reproduce the results, Do I only need to run a train_{algorithm}_{model}_{task} function and run a test_{algorithm}_{model}_{task} function, and then see the result.json?

Yes, your understanding of the process is correct. The "result.json" indeed contains the final results, similar to the metrics presented in the paper.

**Comment by cheng-siyuan** [comment]
Hello, we appreciate your attention to detail in identifying the bugs in our work. We have since updated our experimental results accordingly. The latest findings are now available for review [here](https://arxiv.org/pdf/2310.08475.pdf). Note that these updates do not alter the overall experimental conclusions presented in our paper.

**Comment by cheng-siyuan** [comment]
If you have any further questions, please feel free to submit issues.
